scripts/scil_summarize_bundles_similarity_measures.py

```diff
@@ -0,0 +1,328 @@
+                         disable_centroids=False):
+    # Since data is often re-use when comparing multiple bundles, anything
+    # that can be computed once is saved temporarily and simply loaded on demand
+    if not os.path.isfile(filename):
+        if init_only:
+            print('{0} does not exist'.format(filename))
+        return None
+
+    hash_tmp = hashlib.md5(filename.encode()).hexdigest()
+    tmp_density_filename = os.path.join('tmp_measures/',
+                                        '{0}_density.nii.gz'.format(hash_tmp))
+    tmp_endpoints_filename = os.path.join('tmp_measures/',
+                                          '{0}_endpoints.nii.gz'.format(hash_tmp))
+    tmp_centroids_filename = os.path.join('tmp_measures/',
+                                          '{0}_centroids.trk'.format(hash_tmp))
+
+    sft = load_tractogram(filename, reference,
+                          to_space=Space.VOX,
+                          shifted_origin=True)
+    streamlines = sft.get_streamlines_copy()
+    if not streamlines:
+        if init_only:
+            print('{0} is empty'.format(filename))
+        return None
+
+    if os.path.isfile(tmp_density_filename) \
+            and os.path.isfile(tmp_endpoints_filename) \
+            and os.path.isfile(tmp_centroids_filename):
+        # If initilization, loading the data is useless
+        if init_only:
+            return None
+        density = nib.load(tmp_density_filename).get_data()
+        endpoints_density = nib.load(tmp_endpoints_filename).get_data()
+        sft_centroids = load_tractogram(tmp_centroids_filename, reference,
+                                        to_space=Space.VOX,
+                                        shifted_origin=True)
+        centroids = sft_centroids.get_streamlines_copy()
+    else:
+        transformation, dimensions, _, _ = sft.space_attribute
+        density = compute_tract_counts_map(streamlines, dimensions)
+        endpoints_density = get_endpoints_map(streamlines, dimensions)
+        thresholds = [32, 24, 12, 6]
+        if disable_centroids:
+            centroids = []
+        else:
+            centroids = qbx_and_merge(streamlines, thresholds,
+                                      rng=RandomState(0),
+                                      verbose=False).centroids
+
+        # Saving tmp files to save on future computation
+        nib.save(nib.Nifti1Image(density.astype(np.float32), transformation),
+                 tmp_density_filename)
+        nib.save(nib.Nifti1Image(endpoints_density.astype(np.int16),
+                                 transformation),
+                 tmp_endpoints_filename)
+
+        centroids_sft = StatefulTractogram(centroids, reference, Space.VOX,
+                                           shifted_origin=True)
+        save_tractogram(centroids_sft, tmp_centroids_filename)
+
+    return (density, endpoints_density, streamlines, centroids)
+
+
+def compute_all_measures(args):
```

Comment:
is it required to have only one argument like this here ? I don't see why they are not named (even with Pool it should be working)

Reply:
Pool seems to be expecting a single `args`, the alternative online seems to be either what I did or to add a third function simply to pass its single `args` to another function with `*args`(which seems unnecessary to me)